judgments/forms/search_forms.py

```diff
@@ -21,20 +24,46 @@ def _get_choices_by_group(courts):
         "key2": "value2"
     }
     """
-    options: dict = {}
+    options: dict[Union[Optional[str], CourtParam], Union[str, dict[CourtParam, str]]] = {}
     for group in courts:
         if group.display_heading:
-            option = {group.name: {court.canonical_param: court.grouped_name for court in group.courts}}
+            option_a: dict[Union[Optional[str], CourtParam], Union[str, dict[CourtParam, str]]] = {
```

Comment:
These `option_a` and `option_b` names could be clearer, possibly as part of moving the `options` type definition into an alias so it's clearer what's going on here.

Reply:
Renamed to isolated_court and court_group